### PR TITLE
fix: ROCm build

### PR DESCRIPTION
**Pull request by Naomiusearch**
I have no idea what to put here so...

Pozdrawiam z rodzinką :3

### CODE REVIEW

File: amdpatch.sh [hunk 1]
```diff
@@ -2,4 +2,4 @@
 
 ROCM_PATH=$(hipconfig --rocmpath)
 
-sudo patch $ROCM_PATH/lib/llvm/lib/clang/18/include/__clang_hip_cmath.h ./patches/amd.patch
\ No newline at end of file
+sudo patch $ROCM_PATH/lib/llvm/lib/clang/*/include/__clang_hip_cmath.h ./patches/amd.patch
```
Comment (AlpinDale) [inline amdpatch.sh:5]: What happens if there's multiple versions installed here? Will it return the first one?

File: CMakeLists.txt [hunk 2]
```diff
@@ -221,6 +219,10 @@ if(APHRODITE_GPU_LANG STREQUAL "CUDA")
     "kernels/all_reduce/custom_all_reduce.cu")
 
   # Add CUTLASS and GPTQ Marlin kernels if not MSVC
```
Comment (AlpinDale) [inline CMakeLists.txt:221]: This comment might be inappropriate here

**Review by AlpinDale — APPROVED** [review]
LGTM, just a few nitpicks

File: amdpatch.sh (hunk 1 above)
Comment (Naomiusearch) [reply amdpatch.sh:5]: I think there shouldn't be more than one version, though I can just make it check rocm version and patch clang 17 for rocm 6.1 and clang 18 for 6.2 if that's better

File: CMakeLists.txt (hunk 2 above)
Comment (Naomiusearch) [reply CMakeLists.txt]: Missed it, it's fixed now

File: CMakeLists.txt [hunk 3]
```diff
@@ -65,21 +65,20 @@ endif()
...
+  find_package(CUDAToolkit REQUIRED)
+  # Add cuBLAS to the list of libraries to link against
+  list(APPEND LIBS CUDA::cublas)
+  set(CMAKE_CXX_STANDARD 17)
+  set(CMAKE_CXX_STANDARD_REQUIRED ON)
+  set(CMAKE_CUDA_STANDARD 17)
+  set(CMAKE_CUDA_STANDARD_REQUIRED ON)
+  # Replace -std=c++20 with -std=c++17 in APHRODITE_GPU_FLAGS
+  if(APHRODITE_GPU_LANG STREQUAL "CUDA")
   list(APPEND APHRODITE_GPU_FLAGS "--std=c++17" "-Xcompiler -Wno-return-type")
```
Comment (AlpinDale) [inline CMakeLists.txt:79]:
```suggestion
    list(APPEND APHRODITE_GPU_FLAGS "--std=c++17" "-Xcompiler -Wno-return-type")
```

File: CMakeLists.txt [hunk 4]
```diff
@@ -65,20 +65,19 @@ endif()
...
+  # Add cuBLAS to the list of libraries to link against
+  list(APPEND LIBS CUDA::cublas)
+  set(CMAKE_CXX_STANDARD 17)
+  set(CMAKE_CXX_STANDARD_REQUIRED ON)
+  set(CMAKE_CUDA_STANDARD 17)
+  set(CMAKE_CUDA_STANDARD_REQUIRED ON)
+  # Replace -std=c++20 with -std=c++17 in APHRODITE_GPU_FLAGS
+  if(APHRODITE_GPU_LANG STREQUAL "CUDA")
+    list(APPEND APHRODITE_GPU_FLAGS "--std=c++17" "-Xcompiler -Wno-return-type")
+endif()
```
Comment (AlpinDale) [inline CMakeLists.txt:80]:
```suggestion
  endif()
```

File: amdpatch.sh (hunk 1 above)
Comment (Naomiusearch) [reply amdpatch.sh:5]: Nvm, I'm too stupid for that. It should be fine as is anyway

**Comment by Naomiusearch** [comment]
@AlpinDale can it be merged?

**Comment by AlpinDale** [comment]
@Naomiusearch sorry for the delay. I forgot to mention: I tested this PR a bit more and it seems to break AWQ/GPTQ marlin entirely. I planned to redo this PR but I got distracted. I'll finish it up today.

**Comment by AlpinDale** [comment]
I tested this some more and AWQ needs more work to be fixed. Can you undo the changes related to gptq_marlin? We can merge the PR then.

**Review by AlpinDale — APPROVED** [review]

**Pull request merged**